### PR TITLE
Improve MoveToHead command

### PR DESCRIPTION
**Pull request by pclouds**
The first is related to 624287a (Resize fullscreen/maximized windows
on randr changes - 2022-01-28). The same behavior change is made: if
we move a window to a different head and it's fullscreen (or
maximized) then we make sure it fills the head.

The second is to make MoveToHead accept a direction keyword in
addition to head number. This makes it possible to say "move this
window to the left monitor", which could be bound to a key.

### CODE REVIEW

File: src/Frame.cc [hunk 1]
```diff
@@ -1397,6 +1397,46 @@ Frame::recalcResizeDrag(int nx, int ny, bool left, bool top)
 	_gm.y = top ? (_click_y - _gm.height) : _click_y;
 }
 
+void
+Frame::moveToHead(const std::string& arg)
+{
+	char* endp = NULL;
+	uint head_nr = strtol(arg.c_str(), &endp, 10);
```
Comment (pekdon) [inline src/Frame.cc:1404]: Try using std::stoi instead.

File: src/Frame.cc [hunk 2]
```diff
@@ -1397,6 +1397,46 @@ Frame::recalcResizeDrag(int nx, int ny, bool left, bool top)
...
+	if (endp && *endp == '\0') {
+		moveToHead(head_nr);
+		return;
+	}
+	if (StringUtil::ascii_ncase_equal(arg, "LEFT")) {
```
Comment (pekdon) [inline src/Frame.cc:1409]: I wold much rather have this as an if else if else where the new head number is calculated, and at the end do if (head_nr == -1) { P_ERR() } else { moveToHead(head_nr); } getting rid of all the returns and moveToHead calls.

File: src/X11Util.hh [hunk 3]
```diff
@@ -64,6 +64,7 @@ public:
 namespace X11Util {
 	Geometry getHeadGeometry(int head);
 	uint getCurrHead(CurrHeadSelector chs);
+	bool getNearestHead(const PWinObj& wo, uint& nearest_head, int* dx = NULL, int* dy = NULL);
```
Comment (pekdon) [inline src/X11Util.hh:67]: s/NULL/nullptr/

File: src/X11.cc [hunk 4]
```diff
@@ -813,21 +814,48 @@ X11::getScreenChangeNotification(XEvent *ev, ScreenChangeNotification &scn)
...
+
+//! @brief Searches for the head closest to the coordinates x,y.
+//!
+//! *dx and *dy contain direction values -1 (west or north), 0 and 1
+//! (east or south). If not null, only the nearest window in that
+//! direction is selected.
+//!
+//! @return The nearest head.  Head numbers are indexed from 0.
+bool
+X11::getNearestHead(int x, int y, uint& nearest_head, int* dx, int* dy)
```
Comment (pekdon) [inline src/X11.cc:833]: s/int/enum DirectionType/g to make it a bit more clear.

Also, could we change from a pointer and instead add another value to the type as the skip value?

File: src/Frame.cc [hunk 5]
```diff
@@ -1397,6 +1397,35 @@ Frame::recalcResizeDrag(int nx, int ny, bool left, bool top)
...
+	int head_nr;
+	try {
+		head_nr = std::stoi(arg);
+	} catch (std::invalid_argument&) {
+		head_nr = -1;
+	}
+
+	if (head_nr != -1) {
+		// valid value from stoi above
+	} else if (StringUtil::ascii_ncase_equal(arg, "LEFT")) {
```
Comment (pclouds) [inline src/Frame.cc:1412]: Not sure if you like a flatter code flow like this, or move this if/else chain inside the catch block.

**Comment by pekdon** [comment]
Thanks for feature improvement @pclouds!

**Pull request merged**